### PR TITLE
[GOBBLIN-1273] Improving handling of config file load failures

### PR DESCRIPTION
**Pull request by shirshanka**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1273

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
  - Adds line number and file name of file where error was encountered, if that information is retrievable from the thrown exception
  - Prevents config file loading failures from escaping and aborting the Gobblin standalone service. The service will continue to operate with the config files that it was able to load. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
 - Adds test in PullFileLoaderTest to check for line number in the exception trace

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java [hunk 1]
```diff
@@ -284,43 +287,62 @@ private Config findAndLoadGlobalConfigInDirectory(Path path, Config fallback) th
...
+   * Wrap a {@link ConfigException} (which extends {@link RuntimeException} with an IOException,
+   * with a helpful message if possible
+   * @param path
+   * @param configException
+   * @return an {@link IOException} wrapping the passed in ConfigException.
+   */
+  private IOException wrapConfigException(Path path, ConfigException configException) {
+    if (configException.origin() != null) {
+      return new IOException("Failed to parse config file " + path.toString()
+          + " at lineNo:" + configException.origin().lineNumber(),configException);
```
Comment (sv2000) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java:332]: Nit: Add a whitespace after the ",".

File: gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java [hunk 2]
```diff
@@ -175,14 +176,14 @@ public Config apply(@Nullable Path jobFile) {
         }
 
         try {
-          return PullFileLoader.this.loadPullFile(jobFile,
-              sysProps, loadGlobalProperties);
-        } catch (IOException e) {
-          log.error("Cannot load job from {} due to {}", jobFile, ExceptionUtils.getFullStackTrace(e));
+          return PullFileLoader.this.loadPullFile(jobFile, sysProps, loadGlobalProperties);
+        } catch (IOException ie) {
+          log.error("",ie);
```
Comment (sv2000) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java:181]: Add white space after ",".

File: gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java [hunk 3]
```diff
@@ -175,14 +176,14 @@ public Config apply(@Nullable Path jobFile) {
...
           return null;
         }
       }
-    });
+    }).stream().filter(x -> x!= null).collect(Collectors.toList());
```
Comment (sv2000) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java:185]: Add whitespaces around "!=".

File: gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java [hunk 4]
```diff
@@ -175,14 +176,14 @@ public Config apply(@Nullable Path jobFile) {
...
+    // only return valid parsed configs
```
Comment (sv2000) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java:186]: Can this behavior be documented in the javadoc of the method?

**Review by sv2000 — CHANGES_REQUESTED** [review]

**Comment by shirshanka** [comment]
@sv2000 : addressed your comments, PTAL

**Review by sv2000 — APPROVED** [review]
+1. LGTM.

**Pull request closed without merging**